### PR TITLE
Collaborative Experiment Tracking Docs 

### PR DESCRIPTION
**Pull request by rashidakanchwala**
[Reviewers: Please see the built docs here](https://kedro--2589.org.readthedocs.build/en/2589/)

## Description
This PR is to write docs on 
- how to set up experiment tracking on Kedro-viz?
- how does collaborative experiment tracking work on Kedro? 

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes

### CODE REVIEW

File: docs/source/visualisation/experiment_tracking.md [hunk 1]
```diff
@@ -11,6 +11,8 @@ The metadata you store may include:
 * Model weights
 * Plots and other visualisations
 
+Kedro-Viz provides a reliable and convenient way to store and access experiment data. With collaborative experiment tracking in Kedro-Viz, you can easily share your experiments with team members. This is made possible through local and remote storage solutions. Whether you choose to store your experiments locally or leverage cloud storage options like AWS S3, Kedro-Viz ensures that all team members can access and analyze their own experiments as well as those conducted by others.
```
Comment (stichbury) [inline docs/source/visualisation/experiment_tracking.md:14]:
```suggestion
You can use Kedro-Viz experiment tracking to store and access results, and to share them with others for comparison. Storage can be local or remote, such as cloud storage on AWS S3.
```

File: docs/source/visualisation/experiment_tracking.md [hunk 2]
```diff
@@ -86,12 +90,41 @@ SESSION_STORE_CLASS = SQLiteStore
 SESSION_STORE_ARGS = {"path": str(Path(__file__).parents[2] / "data")}
 ```
 
-This specifies the creation of the `SQLiteStore` under the `data/` subfolder, using the `SQLiteStore` setup from your installed Kedro-Viz plugin.
+This specifies the creation of the `SQLiteStore` under the `data/` subfolder, using the `SQLiteStore` setup from your installed Kedro-Viz plugin
```
Comment (stichbury) [inline docs/source/visualisation/experiment_tracking.md:93]:
```suggestion
This specifies the creation of the `SQLiteStore` under the `data` subfolder, using the `SQLiteStore` setup from your installed Kedro-Viz plugin
```

File: docs/source/visualisation/experiment_tracking.md [hunk 3]
```diff
@@ -86,12 +90,41 @@ SESSION_STORE_CLASS = SQLiteStore
 SESSION_STORE_ARGS = {"path": str(Path(__file__).parents[2] / "data")}
 ```
 
-This specifies the creation of the `SQLiteStore` under the `data/` subfolder, using the `SQLiteStore` setup from your installed Kedro-Viz plugin.
+This specifies the creation of the `SQLiteStore` under the `data` subfolder, using the `SQLiteStore` setup from your installed Kedro-Viz plugin
 
 This step is crucial to enable experiment tracking features on Kedro-Viz, as it is the database used to serve all run data to the Kedro-Viz front-end. Once this step is complete, you can either proceed to [set up the tracking datasets](#set-up-experiment-tracking-datasets) or [set up your nodes and pipelines to log metrics](#modify-your-nodes-and-pipelines-to-log-metrics); these two activities are interchangeable, but both should be completed to get a working experiment tracking setup.
 
+To enable collaborative experiment tracking in Kedro-Viz, you can specify a remote path in the SESSION_STORE_ARGS variable, which links to your cloud storage. In Kedro-Viz version 6.2, the only way to set up credentials for accessing your cloud storage is through environment variables.
```
Comment (stichbury) [inline docs/source/visualisation/experiment_tracking.md:97]:
```important
In Kedro-Viz version 6.2, the only way to set up credentials for accessing your cloud storage is through environment variables.
```

Comment (stichbury) [inline docs/source/visualisation/experiment_tracking.md:97]:
```suggestion
To enable collaborative experiment tracking in Kedro-Viz, specify a remote path in the `SESSION_STORE_ARGS` variable, which links to your cloud storage. In Kedro-Viz version 6.2, the only way to set up credentials for accessing your cloud storage is through environment variables. 
```

File: docs/source/visualisation/experiment_tracking.md [hunk 4]
```diff
@@ -86,12 +90,41 @@ SESSION_STORE_CLASS = SQLiteStore
...
+```python
+from kedro_viz.integrations.kedro.sqlite_store import SQLiteStore
+from pathlib import Path
+
+SESSION_STORE_CLASS = SQLiteStore
+SESSION_STORE_ARGS = {"path": str(Path(__file__).parents[2] / "data"), 
+                    "remote_path": "s3://my-bucket-name/path/to/experiments"}
+```
+
+Please ensure you have the necessary credentials set up as shown below:
```
Comment (stichbury) [inline docs/source/visualisation/experiment_tracking.md:108]:
```suggestion
Ensure you have the necessary credentials set up as shown below:
```

File: docs/source/visualisation/experiment_tracking.md [hunk 5]
```diff
@@ -86,12 +90,41 @@ SESSION_STORE_CLASS = SQLiteStore
...
+Ensure you have the necessary credentials set up as shown below:
+
+```bash
+export AWS_ACCESS_KEY_ID="your_access_key_id"
+export AWS_SECRET_ACCESS_KEY="your_secret_access_key"
+export AWS_REGION="your_aws_region"
+
+```
+
+Kedro-viz saves your experiments as SQLite database files on a central cloud storage. To ensure that all users have a unique filename you can set up your `KEDRO_SQLITE_STORE_USERNAME` in the environment variables. By default, Kedro-viz will take your computer user name if this is not specified.
```
Comment (stichbury) [inline docs/source/visualisation/experiment_tracking.md:117]:
```suggestion
Kedro-Viz saves your experiments as SQLite database files on a central cloud storage. To ensure that all users have a unique filename set up your `KEDRO_SQLITE_STORE_USERNAME` in the environment variables. By default, Kedro-Viz will take your computer user name if this is not specified.
```

File: docs/source/visualisation/experiment_tracking.md [hunk 6]
```diff
@@ -86,12 +90,41 @@ SESSION_STORE_CLASS = SQLiteStore
...
+
+```
+
+Kedro-Viz saves your experiments as SQLite database files on a central cloud storage. To ensure that all users have a unique filename set up your `KEDRO_SQLITE_STORE_USERNAME` in the environment variables. By default, Kedro-Viz will take your computer user name if this is not specified.
+
+```bash
+export KEDRO_SQLITE_STORE_USERNAME ="your_unique__username"
+
+```
+By following the above steps and providing the necessary credentials and a unique username, you can enable collaborative experiment tracking in Kedro-Viz.
```
Comment (stichbury) [inline docs/source/visualisation/experiment_tracking.md:123]:
```suggestion

```

**Comment by stichbury** [comment]
I'm happy with this @antonymilne @tynandebold @NeroOkwa Please can you take a look, particularly at:

* New section in Viz documentation
 
image

* Docs have an index page for experiment tracking, where I've added some of the original content of the experiment tracking page. 
* Please also take a look at the additions for collaborative experiment tracking that @rashidakanchwala and I have added together.

**Review by NeroOkwa — APPROVED** [review]
@rashidakanchwala this looks good and clearly explains collaborative experiment tracking and how to get started. 

@stichbury I like the inclusion of the content of the original experiment tracking page for context.

**Review by astrojuanlu — COMMENTED** [review]
Gave this a quick pass, looks great!

As a side not, I'm not a fan of the `experiment_tracking/experiment_tracking.html` URL we're generating here, but I know we follow this kind of structure in other places of our docs (with the index page showing a sneak peek, and then another page showing the actual content). As a reader, I think it might be easy to miss the smol link below the image if one is not paying attention or does not expect a link to be there:

image

The flow gets interrupted, and either the reader needs to click "Next" (which may or may not bring them to the appropriate page, since "Next" is entirely context dependent) or look back on the Table of Contents on the left, which contains two sections with a very similar name:

image

@stichbury I'm wondering if we could give this a closer look, maybe in a separate issue.

**Comment by rashidakanchwala** [comment]
> Gave this a quick pass, looks great!
> 
> As a side note, I'm not a fan of the `experiment_tracking/experiment_tracking.html` URL we're generating here, but I know we follow this kind of structure in other places of our docs (with the index page showing a sneak peek, and then another page showing the actual content). As a reader, I think it might be easy to miss the smol link below the image if one is not paying attention or does not expect a link to be there:
> 
> image
> The flow gets interrupted, and either the reader needs to click "Next" (which may or may not bring them to the appropriate page, since "Next" is entirely context dependent) or look back on the Table of Contents on the left, which contains two sections with a very similar name:
> 
> image
> @stichbury I'm wondering if we could give this a closer look, maybe in a separate issue.

Yeah, i felt the same. I just saw this earlier today.

**Review by astrojuanlu — APPROVED** [review]
This actually looks better to me!

**Comment by rashidakanchwala** [comment]
> Reviewers: Please see the built docs here

In the build-docs, i still see it as two pages but in the github changes, it's all there in one page. So yes, we can go ahead and merge. thanks Jo

I tried creating a new spaceflight project and follow the docs and it worked!

**Comment by stichbury** [comment]
You'll need to do a hard refresh (command-R) and it'll show the single page alone -- I had the same problem initially!
I'll merge it, and it's good to know it's all working great. Thank you! 🏆

**Pull request merged**